File: Structural/Facade/JavaScript/dvdPlayer.js
// dvdPlayer.js

class DVDPlayer {
  turnOn() {
    console.log("DVD Player is ON.");
  }

  turnOff() {
    console.log("DVD Player is OFF.");
  }

  play(movie) {
    console.log(`Playing movie: ${movie}`);
  }
}

module.exports = DVDPlayer;

File: Structural/Facade/Cpp/DVDPlayer.cpp
// DVDPlayer.cpp

#include "DVDPlayer.h"

void DVDPlayer::turnOn() {
  std::cout << "DVD Player is ON." << std::endl;
}

void DVDPlayer::turnOff() {
  std::cout << "DVD Player is OFF." << std::endl;
}

void DVDPlayer::play(const std::string& movie) {
  std::cout << "Playing movie: " << movie << std::endl;
}
